src/main/java/com/contrastsecurity/maven/plugin/ContrastInstallAgentMojo.java

```diff
@@ -1,24 +1,64 @@
 package com.contrastsecurity.maven.plugin;
 
+import com.contrastsecurity.exceptions.UnauthorizedException;
+import com.contrastsecurity.models.AgentType;
+import com.contrastsecurity.models.Applications;
 import com.contrastsecurity.sdk.ContrastSDK;
 import java.io.File;
+import java.io.IOException;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
+import org.apache.commons.io.FileUtils;
 import org.apache.commons.lang.StringUtils;
 import org.apache.maven.model.Plugin;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
 
 @Mojo(name = "install", defaultPhase = LifecyclePhase.VALIDATE, requiresOnline = true)
-public class InstallAgentContrastMavenMojo extends AbstractContrastMavenPluginMojo {
+public class ContrastInstallAgentMojo extends AbstractAssessMojo {
```

Comment:
Refactored to extend `AbstractAssessMojo`. Migrated code from base class that is only used by this class (e.g. the download agent code)